### ISSUE TITLE
Setup private chain

### ISSUE THREAD

**Issue by artjoma** [prysmaticlabs/prysm#14657]
According to docs https://docs.prylabs.network/docs/advanced/proof-of-stake-devnet user can setup private devnet. How to setup private full functional chain with custom genesis, 2-3 validators? Can somebody advice please.  

```
$PRYSM_BIN_FOLDER/prysmctl testnet generate-genesis \
--fork deneb \
--num-validators 0 \
--genesis-time-delay 10 \
--chain-config-file config.yml \
--geth-genesis-json-in genesis.json \
--geth-genesis-json-out genesis.json \
--output-ssz genesis.ssz \
--deposit-json-file="keys/deposit_data-1732289365.json"
```
```
INFO[0000] Setting fork geth times                       cancun=1732496794 prague=7853820991 prefix=genesis shanghai=1732496794
FATA[0000] Could not generate beacon chain genesis state  error="could not generate genesis state: state has nil validator slice" prefix=genesis
```

**Comment by TrungBD2605** [comment]
I have the same question.

**Comment by netzulo** [comment]
i resolved but idk if is interesting to share, various files involved, and... getting 3 full nodes up isn't easy to get

**Comment by TrungBD2605** [comment]
> i resolved but idk if is interesting to share, various files involved, and... getting 3 full nodes up isn't easy to get

Yes I really interested , Could you please share for me. I am struggling for that.

**Comment by ThomasAqu1nas** [comment]
Why is this issue was closed? Got same issue:
➜  consensus git:(DITBCH-257) ✗ ./prysmctl testnet generate-genesis \
        --chain-config-file ../docker/vol/data/chain-config/config.yaml \
        --geth-genesis-json-in ../docker/vol/data/chain-config/genesis.json \
        --geth-genesis-json-out ../docker/vol/data/chain-config/genesis.json \
        --output-ssz ../docker/vol/data/chain-config/genesis.ssz --config-name=custom --fork=electra --deposit-json-file=/Users/ando/Documents/work/projects/ag2/ag-mos-ru-ethereum/ag-migration/network/docker/vol/data/chain-config/deposits.json --override-eth1data --num-validators=3
INFO[0000] Specified a chain config file: ../docker/vol/data/chain-config/config.yaml  prefix=genesis
INFO[0000] No genesis time specified, defaulting to now()  prefix=genesis
INFO[0000] Delaying genesis 1738250089 by 0 seconds      prefix=genesis
INFO[0000] Genesis is now 1738250089                     prefix=genesis
INFO[0000] reading deposits from JSON at /Users/ando/Documents/work/projects/ag2/ag-mos-ru-ethereum/ag-migration/network/docker/vol/data/chain-config/deposits.json  prefix=genesis
INFO[0000] Setting fork geth times                       cancun=1738250089 prague=1738250089 prefix=genesis shanghai=1738250089
FATA[0000] Could not generate beacon chain genesis state  error="could not generate genesis state: state has nil validator slice" prefix=genesis

where deposits.json contains the list of 3 generated with eth2-staking-deposit binary deposit json's for 3 different mnemonics

**Comment by syjn99** [comment]
BTW, have you ever tried [ethereum-package](https://github.com/ethpandaops/ethereum-package) using kurtosis? In my own experience, it's more seamless.

https://x.com/jameshe_eth/status/1879647509179752518

**Comment by ThomasAqu1nas** [comment]
> BTW, have you ever tried [ethereum-package](https://github.com/ethpandaops/ethereum-package) using kurtosis? In my own experience, it's more seamless.
> 
> https://x.com/jameshe_eth/status/1879647509179752518

I've heard about this package, but it's not the right solution for me.
